Source/SceneEditor/SceneEditor.cpp

```diff
@@ -264,6 +251,311 @@ namespace PlatinumEngine{
+					//----------------------------------------------------------------//
+
+					Maths::Vec3 vertex0, vertex1, vertex2;
+
+					// check if the game object has transformation components
+					if (auto transformComponent = currentCheckingGameobject->GetComponent<TransformComponent>();
+							transformComponent != nullptr)
+					{
+						Maths::Mat4 translationMatrix, rotationMatrix, scaleMatrix;
+						translationMatrix.SetTranslationMatrix(transformComponent->position);
+						rotationMatrix.SetRotationMatrix(transformComponent->rotation.EulerAngles());
+						scaleMatrix.SetScaleMatrix(transformComponent->scale);
+
+						Maths::Vec4 temporaryMatrix;
+
+						// get the world coordinate of vertex 0
+						temporaryMatrix = rotationMatrix * translationMatrix * scaleMatrix *
+										  Maths::Vec4(mesh->vertices[mesh->indices[count + 0]].position.x,
+												  mesh->vertices[mesh->indices[count + 0]].position.y,
+												  mesh->vertices[mesh->indices[count + 0]].position.z, 1.0f);
+						temporaryMatrix = temporaryMatrix / temporaryMatrix.w;
+						vertex0 = PlatinumEngine::Maths::Vec3(temporaryMatrix.x, temporaryMatrix.y, temporaryMatrix.z);
+
+						// get the world coordinate of vertex 1
+						temporaryMatrix = rotationMatrix * translationMatrix * scaleMatrix *
+										  Maths::Vec4(mesh->vertices[mesh->indices[count + 1]].position.x,
+												  mesh->vertices[mesh->indices[count + 1]].position.y,
+												  mesh->vertices[mesh->indices[count + 1]].position.z, 1.0f);
+						temporaryMatrix = temporaryMatrix / temporaryMatrix.w;
+						vertex1 = PlatinumEngine::Maths::Vec3(temporaryMatrix.x, temporaryMatrix.y, temporaryMatrix.z);
+
+						// get the world coordinate of vertex 2
+						temporaryMatrix = rotationMatrix * translationMatrix * scaleMatrix *
+										  Maths::Vec4(mesh->vertices[mesh->indices[count + 2]].position.x,
+												  mesh->vertices[mesh->indices[count + 2]].position.y,
+												  mesh->vertices[mesh->indices[count + 2]].position.z, 1.0f);
+						temporaryMatrix = temporaryMatrix / temporaryMatrix.w;
+						vertex2 = PlatinumEngine::Maths::Vec3(temporaryMatrix.x, temporaryMatrix.y, temporaryMatrix.z);
+					}
+					else
+					{
+						vertex0 = mesh->vertices[mesh->indices[count + 0]].position;
+						vertex1 = mesh->vertices[mesh->indices[count + 1]].position;
+						vertex2 = mesh->vertices[mesh->indices[count + 2]].position;
+					}
+
+					//----------------------------------//
+					// Do ray and triangle intersection //
+					//----------------------------------//
+
+					// build the PCS using the triangle primitive
+
+					// origin of PCS
+					Maths::Vec3 originForPCS = vertex0;
+
+					// not normalized axis of PCS
+					Maths::Vec3 axisUForPCS = vertex1 - originForPCS;
+					Maths::Vec3 axisNForPCS = Cross(axisUForPCS, (vertex2 - originForPCS));
+					Maths::Vec3 axisWForPCS = Cross(axisUForPCS, axisNForPCS);
```

Comment:
Axis U N W?

Reply:
Mentioned in intersection PowerPoint for fundamental modeling last semester.